base/static/base/js/jquery.track-everything.js

```diff
@@ -63,362 +29,572 @@
-                params.event_category = params.event_category || CATEGORIES.SHORTCUTS;
-                params.event_subcategory = params.event_subcategory || CATEGORIES.SHORTCUTS;
-            }
-            // footer
-            else if (link.closest(SELECTORS.FOOTER)) {
-                params.event_category = params.event_category || CATEGORIES.FOOTER;
-                const listParent = link.closest('ul, ol');
-                params.event_subcategory = params.event_subcategory ||
-                    (listParent ? listParent.getAttribute('aria-labelledby') : null) || CATEGORIES.FOOTER;
-            }
-            // any widget
-            else if (link.closest(SELECTORS.WIDGET)) {
-                params.event_category = params.event_category || CATEGORIES.WIDGET;
-                const widgetParent = link.closest(SELECTORS.WIDGET);
-                params.event_subcategory = params.event_subcategory ||
-                    (widgetParent && widgetParent.id && widgetParent.id.includes('widget') ? widgetParent.id : null) || null;
-                if (link.closest(SELECTORS.SEARCH_WIDGET)) {
-                    params.event_indecision_count = link.closest(SELECTORS.SEARCH_WIDGET).getAttribute('data-ga-indecision-count') || null;
-                }
-            }
-            // any sidebar
-            else if (link.closest(SELECTORS.SIDEBAR)) {
-                params.event_category = params.event_category || CATEGORIES.SIDEBAR;
-                const sidebarParent = link.closest(SELECTORS.SIDEBAR);
-                params.event_subcategory = params.event_subcategory ||
-                    (sidebarParent && sidebarParent.id && sidebarParent.id.includes('sidebar') ? sidebarParent.id : '') ||
-                    (sidebarParent && sidebarParent.className && sidebarParent.className.split(' ').find(function (c) { return c.includes('sidebar'); }) || null);
-            }
-            // Catalog Vufind Search results
-            else if (window.location.href.indexOf(LOCATIONS.VUFIND) > -1) {
-                params.event_category = params.event_category || CATEGORIES.VUFIND_RESULTS;
-                params.event_subcategory = params.event_subcategory ||
-                    link.closest('.action-toolbar') ? 'Action Toolbar' :
-                    link.closest('.searchtools') ? 'Searchtools' :
-                        link.closest('.pagination') ? 'Pagination' :
-                            link.closest('[id]').getAttribute('id');
-                params.event_label = link.classList.contains('title') ? 'Title' :
-                    link.classList.contains('result-author') ? 'Author' :
-                        link.classList.contains('external') ? 'Holding' :
-                            link.classList.contains('save-record') ? 'Save Record' :
-                                params.event_label || 'Unknown';
+            if (rule.location && window.location.href.indexOf(rule.location) === -1) { return; }

```

Comment:
Using `includes` in place of `indexOf` for these  `window.location.href` tests, might be a little cleaner and easier to read. `includes` is used for substring matching. Not a big deal if you want to go with `indexOf` though. I'm just throwing it out there:

```
if (rule.location && window.location.href.includes(rule.location) === -1) { return; }
```